File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/service/CreateUserResponse.kt
package com.example.service

import com.example.service.CreateUserResponse.Error
import com.example.service.CreateUserResponse.Success
import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonSubTypes.Type
import com.fasterxml.jackson.annotation.JsonTypeInfo

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@type")
@JsonSubTypes(value = [Type(value = Success::class, name = "Success"), Type(value = Error::class, name = "Error")])
/** Response wrapper for createUser RPC call */
sealed interface CreateUserResponse {
  /** Error response */
  data class Error(
    val correlationId: kotlin.String,
    val error: ValidationError
  ) : CreateUserResponse

  /** Successful response */
  data class Success(
    val correlationId: kotlin.String,
    val value: User
  ) : CreateUserResponse
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/service/CreateUserResponse.scala
package com.example.service



/** Response wrapper for createUser RPC call */
sealed trait CreateUserResponse {
  def correlationId: String
}

object CreateUserResponse {
  /** Error response */
  case class Error(
    correlationId: String,
    error: ValidationError
  ) extends CreateUserResponse

  /** Successful response */
  case class Success(
    correlationId: String,
    value: User
  ) extends CreateUserResponse
}